api/turing/imagebuilder/imagebuilder.go

```diff
@@ -366,7 +366,8 @@ func getGCPSubDomains() []string {
 }
 
 func (ib *imageBuilder) checkIfImageExists(imageName string, imageTag string) (bool, error) {
-	keychain := authn.DefaultKeychain
+	var keychain authn.Keychain
+	keychain = authn.DefaultKeychain
```

Comment:
The reasoning behind this change is exactly the same as the one given in this comment: https://github.com/caraml-dev/merlin/pull/551#discussion_r1521027966